plugins/opensslcryptoplugin/evp_p.h

```diff
@@ -232,11 +232,57 @@ int osslevp_aes_decrypt_ciphertext(const unsigned char *init_vector,
     *decrypted = plaintext;
 
     /* Clean up the decryption context */
-    EVP_CIPHER_CTX_cleanup(&decryption_context);
+    EVP_CIPHER_CTX_free(decryption_context);
     plaintext_length = update_length + final_length;
     return plaintext_length;
 }
 
+/*
+    Implements signing according to:
+    https://wiki.openssl.org/index.php/EVP_Signing_and_Verifying
+ */
+int osslevp_sign(const EVP_MD *digestFunc, EVP_PKEY *pkey, const void *bytes, size_t bytesCount, uint8_t **signature, size_t *signatureLength) {
+    EVP_MD_CTX *mdctx = EVP_MD_CTX_create();
+    int r;
```

Comment:
combine the declaration with the first use, i.e. int r = EVP_DigestSignInit(...)

Reply:
I usually don't, when the variable is then used for different stuff repeatedly. But I'll change it.